platform/api/src/api/v1/gql/models/channel.rs

```diff
@@ -75,15 +81,142 @@ impl<G: ApiGlobal> Channel<G> {
 
 		Ok(followers)
 	}
+
+	async fn live_viewer_count(&self, ctx: &Context<'_>) -> Result<i32> {
+		let global = ctx.get_global::<G>();
+
+		if let Some(count) = self.live_viewer_count_ {
+			let expired = self
+				.live_viewer_count_updated_at
+				.as_ref()
+				.map(|DateRFC3339(t)| Utc::now().signed_duration_since(t).num_seconds() > 30)
+				.unwrap_or(true);
+			if !expired {
+				return Ok(count);
+			}
+		}
+
+		let res = global
+			.video_playback_session_client()
+			.clone()
+			.count(pb::scuffle::video::v1::PlaybackSessionCountRequest {
+				filter: Some(pb::scuffle::video::v1::playback_session_count_request::Filter::Target(
+					pb::scuffle::video::v1::types::PlaybackSessionTarget {
+						target: Some(pb::scuffle::video::v1::types::playback_session_target::Target::RoomId(
+							self.room_id.0.into(),
+						)),
+					},
+				)),
+			})
+			.await
+			.map_err_gql("failed to fetch playback session count")?;
+
+		let live_viewer_count = res.into_inner().deduplicated_count as i32; //should be safe to cast
+
+		sqlx::query(
+			"UPDATE users SET channel_live_viewer_count = $1, channel_live_viewer_count_updated_at = NOW() WHERE id = $2",
+		)
+		.bind(live_viewer_count)
+		.bind(self.id.to_uuid())
+		.execute(global.db().as_ref())
+		.await
+		.map_err_gql("failed to update live viewer count")?;
+
+		Ok(live_viewer_count)
+	}
+}
+
+#[derive(Clone)]
+pub struct ChannelLive<G: ApiGlobal> {
+	// Note: This is not a SimpleObject, for the gql type definition see the impl block below
+	room_id: ulid::Ulid,
+	_phantom: std::marker::PhantomData<G>,
+}
+
+#[derive(serde::Serialize)]
+struct PlayerTokenClaim {
+	organization_id: String,
+	room_id: String,
+	iat: i64,
+	user_id: String,
+}
+
+#[Object]
+impl<G: ApiGlobal> ChannelLive<G> {
+	async fn live(&self, ctx: &Context<'_>) -> Result<bool> {
+		let global = ctx.get_global::<G>();
+
+		let res = global
+			.video_room_client()
+			.clone()
+			.get(pb::scuffle::video::v1::RoomGetRequest {
+				ids: vec![self.room_id.into()],
+				..Default::default()
+			})
+			.await
+			.map_err_gql("failed to fetch room")?;
+		let room = res
+			.into_inner()
+			.rooms
+			.into_iter()
+			.next()
+			.map_err_gql("failed to fetch room")?;
+
+		Ok(room.status == pb::scuffle::video::v1::types::RoomStatus::Ready as i32)
```

Comment:
We should know this without querying the video api. It should be in the DB since we consume the events.

Reply:
Infact this should not be here.